Content.Server/GameObjects/Components/Nutrition/CigSmokingComponent.cs

```diff
@@ -0,0 +1,141 @@
+    /// <item><description>  you need to have cigar equipped as mask.</description></item>
+    ///</list>
+    /// TODO: Add reagents that interact when smoking
+    /// TODO: Allow suicide via excessive Smoking
+    /// TODO: E-cigarettes
+    /// </summary>
+    [RegisterComponent]
+    public class CigSmokingComponent : SharedSmokingComponent, IInteractUsing, IHotItem
+    {
+        private SmokingStates _currentState;
+        private ClothingComponent _clothingComponent;
+
+        /// <summary>
+        /// Determines the type of cigarette.
+        /// <list type="bullet">
+        /// <item><term>true</term><description>Then this is an e-cig which can
+        /// be turned on/off without needing to be light up.</description></item>
+        /// <item><term>false</term><description>then the cigarette will need to be set alight
+        /// with a heat source. And it will slowly drain until it becomes an unusable.</description></item>
+        /// </list>
+        /// </summary>
+        private bool _electric;
+
+        /// <summary>
+        /// Duration represents how long will this item last.
+        /// Generally it ticks down whether it's time-based
+        /// or consumption-based.
+        /// </summary>
+        [ViewVariables(VVAccess.ReadOnly)]
+        private int _duration;
+
+        /// <summary>
+        /// What is the temperature of the cigar?
+        /// <list type="bullet">
+        /// <item><description>Usually for e-cigs it's around 40-50°C</description></item>
+        /// <item><description>For a regular cigar, the temp approaches around 400°C or 580°C
+        /// dependant on where you measure.</description></item>
+        /// </list>
+        /// </summary>
+        [ViewVariables]
+        private float _temperature;
+
+        [ViewVariables]
+        public SmokingStates CurrentState
+        {
+            get => _currentState;
+            private set
+            {
+                _currentState = value;
+
+                switch (_currentState)
+                {
+                    case SmokingStates.Lit:
+                        _clothingComponent.EquippedPrefix = "lit";
+                        _clothingComponent.ClothingEquippedPrefix = "lit";
+                        break;
+                    default:
+                        _clothingComponent.EquippedPrefix = "unlit";
+                        _clothingComponent.ClothingEquippedPrefix = "unlit";
+                        break;
+                }
+
+                if (Owner.TryGetComponent(out AppearanceComponent appearance))
+                {
+                    appearance.SetData(SmokingVisuals.Smoking, _currentState);
+                }
+            }
+        }
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            Owner.TryGetComponent(out _clothingComponent);
```

Comment:
EnsureComponent, though now that I think about it there's the new ComponentDependency stuff but I haven't sussed it out.